fs/sync/sync_test.go

```diff
@@ -977,6 +978,35 @@ func TestSyncWithUpdateOlder(t *testing.T) {
 	fstest.CheckItems(t, r.Fremote, oneO, twoF, threeO, fourF, fiveF)
 }
 
+// Test with a max transfer session
+func TestSyncWithMaxSession(t *testing.T) {
+	r := fstest.NewRun(t)
+	defer r.Finalise()
+
+	maxSession, err := time.ParseDuration("2s")
+	require.NoError(t, err)
+
+	fs.Config.MaxSession = maxSession
+
+	testFiles := make([]fstest.Item, 10000)
```

Comment:
Making 10000 files will kill the integration tests with other backends for certain! (These tests are run against all the backends once per day).

How about just using a few filesa nd temporarily setting a bwlimit to 1b/s or something like that?

Reply:
They're still flaky on my Mac - I thought I'd got down to only needing 5 test files but sometimes it can get through a few hundred :-(